validator/attester/service.go

```diff
@@ -22,49 +26,78 @@ type Attester struct {
 	cancel         context.CancelFunc
 	assigner       assignmentAnnouncer
 	assignmentChan chan bool
+	p2p            shared.P2P
+	blockBuf       chan p2p.Message
 }
 
 // Config options for an attester service.
 type Config struct {
-	AssignmentBuf int
-	Assigner      assignmentAnnouncer
+	AssignmentBuf   int
+	Assigner        assignmentAnnouncer
+	BlockBufferSize int
 }
 
 // NewAttester creates a new attester instance.
-func NewAttester(ctx context.Context, cfg *Config) *Attester {
+func NewAttester(ctx context.Context, cfg *Config, validatorP2P shared.P2P) *Attester {
 	ctx, cancel := context.WithCancel(ctx)
 	return &Attester{
 		ctx:            ctx,
 		cancel:         cancel,
 		assigner:       cfg.Assigner,
 		assignmentChan: make(chan bool, cfg.AssignmentBuf),
+		p2p:            validatorP2P,
+		blockBuf:       make(chan p2p.Message, cfg.BlockBufferSize),
 	}
 }
 
 // Start the main routine for an attester.
-func (p *Attester) Start() {
+func (a *Attester) Start() {
 	log.Info("Starting service")
-	go p.run(p.ctx.Done())
+	go a.run(a.ctx.Done())
 }
 
 // Stop the main loop.
-func (p *Attester) Stop() error {
-	defer p.cancel()
+func (a *Attester) Stop() error {
+	defer a.cancel()
 	log.Info("Stopping service")
 	return nil
 }
 
+// CreateAttestation will cause the attester to attest to the provided beacon block.
+func (a *Attester) CreateAttestation(block *pbp2p.BeaconBlock) *pbp2p.AttestationRecord {
+	attestation := &pbp2p.AttestationRecord{}
+	attestation.Slot = block.GetSlotNumber()
+
+	// TODO(#487): Attesters responsibilities will be handled by this PR(#487).
+	attestation.AttesterBitfield = []byte{}
+
+	return attestation
+}
+
 // run the main event loop that listens for an attester assignment.
-func (p *Attester) run(done <-chan struct{}) {
-	sub := p.assigner.AttesterAssignmentFeed().Subscribe(p.assignmentChan)
+func (a *Attester) run(done <-chan struct{}) {
+	sub := a.assigner.AttesterAssignmentFeed().Subscribe(a.assignmentChan)
+	blocksub := a.p2p.Subscribe(&shardingp2p.BlockBroadcast{}, a.blockBuf)
 	defer sub.Unsubscribe()
+	defer blocksub.Unsubscribe()
+
 	for {
 		select {
 		case <-done:
 			log.Debug("Attester context closed, exiting goroutine")
 			return
-		case <-p.assignmentChan:
+		case <-a.assignmentChan:
 			log.Info("Performing attester responsibility")
+		case msg := <-a.blockBuf:
```

Comment:
I think this should be in another go routine. If the blockBuf logic is slow then it will end up blocking the assignment logic and vice versa.

I recommend to refactor this method and break out the responsabilities so that it looks something like:
```
func (a *Attester) run(done <- chan struct{}) {
    go a.handleAssignments(done)
    go a.handleBlockBroadcast(done) // Actually, what is this action?
}
```

Reply:
Alright I will break it off for both proposers and attesters into smaller goroutines